include/parameters.h

```diff
@@ -83,7 +83,7 @@ class IndexWriteParametersBuilder
 
     IndexWriteParametersBuilder &with_num_threads(const uint32_t num_threads)
     {
-        _num_threads = num_threads == 0 ? omp_get_num_threads() : num_threads;
+        _num_threads = num_threads == 0 ? omp_get_num_procs() : num_threads;
```

Comment:
I assume this will play nicely with multi-core processors ie 2 quad-core would yield 8 threads.

Reply:
yeah that seems to be the idea - my 12 core i7 shows up as 12 when I use omp_get_num_procs()